python/lsst/analysis/ap/apdb.py

```diff
@@ -66,13 +78,65 @@ def connection(self):
         """
         pass
 
-    def load_sources_for_object(self, dia_object_id, limit=100000):
+    def set_bad_DiaSource_flags(self, flag_list):
+        """Set DiaSource flags to exclude.
+
+        Parameters
+        ----------
+        flag_list : `list` [`str`]
+            Flag names to exclude
+        """
+
+        flag_map = os.path.join(lsst.utils.getPackageDir("ap_association"),
+                                "data/association-flag-map.yaml")
+        unpacker = UnpackApdbFlags(flag_map, "DiaSource")
+
+        for flag in flag_list:
+            if flag not in [c[0] for c in unpacker.output_flag_columns['flags']]:
+                raise ValueError(f"flag {flag} not included in DiaSource flags")
+
+        self.bad_DiaSource_flags = flag_list
+
+    def _make_flag_exclusion_clause(self, flag_list, column_name='flags',
+                                    table_name="DiaSource"):
+        """Create a SQL where clause that excludes sources with chosen flags.
```

Comment:
This kind of thing is making me lean even more towards redoing this in SqlAlchemy. But at least it's all private, so we can go with it as-is for now.

... so many opportunities for injection attacks and malformed queries.

Reply:
I agree that we should consider SqlAlchemy, if only to try to avoid rolling our own stuff like this if we're going to add more complex queries.  

I don't feel like injection attacks are a big practical risk at present--there's nothing that stops someone from grabbing `apdb.connection` and calling `DROP TABLE` directly, no injection required.